// react-uikit-modal/lib/comfirmButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactUikitButton = require('react-uikit-button');

var _reactUikitButton2 = _interopRequireDefault(_reactUikitButton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ComfirmButton = function ComfirmButton(props) {
  if (props.confirm && props.confirm.show) {
    var cleanComfirmProps = _extends({}, props.confirm, { show: null, dialog: null });

    return _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_reactUikitButton2.default, _extends({
        kitid: 'ok-' + props.kitid,
        body: 'cancel',
        margin: 'right'
      }, props.cancel)),
      _react2.default.createElement(_reactUikitButton2.default, _extends({}, cleanComfirmProps, {
        kitid: 'confrim-' + props.kitid,
        body: props.confirm.body || 'Confirm'
      }))
    );
  } else {
    return _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_reactUikitButton2.default, _extends({
        kitid: 'ok-' + props.kitid,
        body: 'cancel',
        margin: 'right'
      }, props.cancel)),
      _react2.default.createElement(_reactUikitButton2.default, _extends({
        kitid: 'cancel-' + props.kitid,
        body: 'ok'
      }, props.ok))
    );
  }
};

ComfirmButton.propTypes = {
  cancel: _react2.default.PropTypes.object,
  confirm: _react2.default.PropTypes.object,
  kitid: _react2.default.PropTypes.string,
  ok: _react2.default.PropTypes.object
};

exports.default = ComfirmButton;
