// react-uikit-form/lib/form-input-main.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _cleanProps = require('./cleanProps');

var _cleanProps2 = _interopRequireDefault(_cleanProps);

var _reactUikitBase = require('react-uikit-base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FormInputMain = function FormInputMain(props) {
  var cssClassNames = _reactUikitBase.helpers.cleanClasses([props.context ? 'uk-form-' + props.context : null, props.size ? 'uk-form-' + props.size : null, props.width ? 'uk-form-width-' + props.width : null, props.blank ? 'uk-form-blank' : null, props.wide ? 'uk-width-1-1' : null, props.className]);

  return _react2.default.createElement('input', _extends({
    id: props.kitid,
    name: props.name
  }, (0, _cleanProps2.default)(props), {
    className: cssClassNames,
    'data-kitid': props.kitid,
    type: props.type || 'text',
    onChange: props.onChange
  }));
};

FormInputMain.propTypes = {
  blank: _react2.default.PropTypes.bool,
  className: _react2.default.PropTypes.string,
  classes: _react2.default.PropTypes.array,
  context: _react2.default.PropTypes.oneOf(['danger', 'success']),
  disabled: _react2.default.PropTypes.bool,
  display: _react2.default.PropTypes.string,
  icon: _react2.default.PropTypes.string,
  kitid: _react2.default.PropTypes.string,
  name: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func,
  required: _react2.default.PropTypes.bool,
  size: _react2.default.PropTypes.oneOf(['large', 'small']),
  type: _react2.default.PropTypes.string,
  value: _react2.default.PropTypes.any,
  wide: _react2.default.PropTypes.bool,
  width: _react2.default.PropTypes.oneOf(['large', 'medium', 'small', 'mini'])
};

exports.default = FormInputMain;
