// react-uikit-form/lib/form-redux-input.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formInput = require('./form-input');

var _formInput2 = _interopRequireDefault(_formInput);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FormReduxInput = function FormReduxInput(field) {
  var blank = field.blank,
      button = field.button,
      context = field.context,
      disabled = field.disabled,
      help = field.help,
      icon = field.icon,
      input = field.input,
      label = field.label,
      onChange = field.onChange,
      options = field.options,
      Password = field.Password,
      placeholder = field.placeholder,
      row = field.row,
      size = field.size,
      toggle = field.toggle,
      type = field.type,
      width = field.width,
      meta = field.meta;


  return _react2.default.createElement(_formInput2.default, _extends({}, input, {
    label: label,
    blank: blank,
    button: button,
    context: context,
    disabled: disabled,
    help: meta.touched && meta.error ? _extends({}, help, { text: meta.error, context: 'danger' }) : null,
    icon: icon,
    options: options,
    Password: Password,
    placeholder: placeholder,
    size: size,
    toggle: toggle,
    type: type,
    width: width,
    row: row
  }));
};

exports.default = FormReduxInput;
